src/engine/math_util.h

```diff
@@ -612,4 +612,6 @@ ALWAYS_INLINE f32 remap(f32 x, f32 fromA, f32 toA, f32 fromB, f32 toB) {
     return (x - fromA) / (toA - fromA) * (toB - fromB) + fromB;
 }
 
+f32 lerpf(f32 start_value, f32 end_value, f32 progress);
```

Comment:
```suggestion
f32 lerpf(f32 start_value, f32 end_value, f32 progress);

ALWAYS_INLINE f32 to_smoothstop(f32 x) {
    f32 sq = sqr(1.0f-x);
    return 1.0f - sq;
}

ALWAYS_INLINE f32 lerp_smoothstart(f32 from, f32 to, f32 amt) {
    return lerpf(from, to, sqr(amt));
}

ALWAYS_INLINE f32 lerp_smoothstop(f32 from, f32 to, f32 amt) {
    return lerpf(from, to, to_smoothstop(amt));
}

ALWAYS_INLINE f32 lerp_smoothstep(f32 from, f32 to, f32 amt) {
    f32 lerpAmt = lerpf(sqr(amt), to_smoothstop(amt), amt);
    return lerpf(from, to, lerpAmt);
}
```

check the pr channel in hacker, we think `lerp_smoothstep` looks a little better here and i think these functions are really useful to have around anyways so might as well have them around